Userland/Libraries/LibWeb/StorageAPI/NavigatorStorage.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024, Jamie Mansfield <jmansfield@cadixdev.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Heap/Heap.h>
+#include <LibJS/Runtime/Realm.h>
+#include <LibWeb/StorageAPI/NavigatorStorage.h>
+
+namespace Web::StorageAPI {
+
+NavigatorStorage::NavigatorStorage(JS::Realm& realm)
+    : m_realm(realm)
+{
+}
+
+// https://storage.spec.whatwg.org/#dom-navigatorstorage-storage
+JS::NonnullGCPtr<StorageManager> NavigatorStorage::storage()
+{
+    if (!m_storage)
+        m_storage = m_realm->heap().allocate<StorageManager>(m_realm, m_realm);
```

Comment:
I think this needs to be stored on the Realm?

[Here:](https://storage.spec.whatwg.org/#dom-navigatorstorage-storage)
> The storage getter steps are to return [this](https://webidl.spec.whatwg.org/#this)’s [relevant settings object](https://html.spec.whatwg.org/multipage/webappapis.html#relevant-settings-object)’s [StorageManager](https://storage.spec.whatwg.org/#storagemanager) object.

and [here:](https://html.spec.whatwg.org/multipage/webappapis.html#relevant-settings-object)

> Then, the relevant settings object for a [platform object](https://webidl.spec.whatwg.org/#dfn-platform-object) o is the [environment settings object](https://html.spec.whatwg.org/multipage/webappapis.html#concept-realm-settings-object) of the [relevant realm](https://html.spec.whatwg.org/multipage/webappapis.html#concept-relevant-realm) for o.

Reply:
Or at least, not in the NavigatorStorage.

Reply:
That sounds like it should be stored in `HTML::EnvironmentSettingsObject` to me.
> Each [environment settings object](https://html.spec.whatwg.org/multipage/webappapis.html#environment-settings-object) has an associated [StorageManager](https://storage.spec.whatwg.org/#storagemanager) object. [[HTML]](https://storage.spec.whatwg.org/#biblio-html)

Reply:
Resolved :^)